packages/content-conformance/src/runner.ts

```diff
@@ -50,14 +55,33 @@ export class ContentConformanceRunner {
     })
   }
 
+  /**
+   * TODO: Determine best ways to surface warnThreshold to user & default warnThreshold
+   */
   async run() {
-    return this.engine?.execute()
+    if (this.status === RunnerStatus.RUNNING) return null
+
+    // @ts-expect-error -- need to sort out VFile types here as well
+    const _statistics = statistics(this.engine?.files)
+
+    this.status = RunnerStatus.RUNNING
+    try {
+      await this.engine?.execute()
+    } catch {
+      this.status = RunnerStatus.FAILURE
+    }
+
+    /**
+     * check vFile-statistics for fatal messages, optionally pass in a warning count threshold
+     * getStatisticsStatus(statistics, warnThreshold)
+     */
+    this.status = getStatisticsStatus(_statistics)
```

Comment:
[nit] I might move `getStatisticsStatus` into the runner class as a private method, just because the logic is pretty specific to the runner.

Reply:
Truth. Will do, thanks!